src/test/java/uk/gov/companieshouse/overseasentitiesapi/validation/UpdateValidatorTest.java

```diff
@@ -70,15 +98,79 @@ void testValidationErrorReportedWhenFilingDateIsInFuture() {
+    void testNoValidationErrorReportedWhenFilingDateIsBeforeNextMadeUpToDate() throws ServiceException {
+        companyProfileApi.getConfirmationStatement().setNextMadeUpTo(LocalDate.now().minusDays(11));
+        when(publicDataRetrievalService.getCompanyProfile(ENTITY_NUMBER, PASS_THROUGH_HEADER)).thenReturn(companyProfileApi);
+        updateDto.setFilingDate(LocalDate.now().minusDays(12));
+
+        Errors errors = updateValidator.validate(ENTITY_NUMBER, updateDto, new Errors(), LOGGING_CONTEXT, PASS_THROUGH_HEADER);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testNoValidationErrorReportedWhenFilingDateIsTheSameAsTheNextMadeUpToDate() throws ServiceException {
+        companyProfileApi.getConfirmationStatement().setNextMadeUpTo(LocalDate.now());
+        when(publicDataRetrievalService.getCompanyProfile(ENTITY_NUMBER, PASS_THROUGH_HEADER)).thenReturn(companyProfileApi);
+        updateDto.setFilingDate(LocalDate.now());
+
+        Errors errors = updateValidator.validate(ENTITY_NUMBER, updateDto, new Errors(), LOGGING_CONTEXT, PASS_THROUGH_HEADER);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testValidationErrorReportedWhenFilingDateIsInThePastButAfterTheNextMadeUpToDate() throws ServiceException {
+        final LocalDate nextMadeUpToDate = LocalDate.now().minusDays(11);
+        companyProfileApi.getConfirmationStatement().setNextMadeUpTo(nextMadeUpToDate);
+        when(publicDataRetrievalService.getCompanyProfile(ENTITY_NUMBER, PASS_THROUGH_HEADER)).thenReturn(companyProfileApi);
+
+        updateDto.setFilingDate(LocalDate.now().minusDays(10));
+        Errors errors = updateValidator.validate(ENTITY_NUMBER, updateDto, new Errors(), LOGGING_CONTEXT, PASS_THROUGH_HEADER);
+
+        String qualifiedFieldName = getQualifiedFieldName(
+                OverseasEntitySubmissionDto.UPDATE_FIELD,
+                UpdateDto.FILING_DATE);
+        String validationMessage = String.format(ValidationMessages.DATE_NOT_ON_OR_BEFORE_MUD_ERROR_MESSAGE, qualifiedFieldName, nextMadeUpToDate);
+        assertError(UpdateDto.FILING_DATE, validationMessage, errors);
+    }
+
+    @Test
+    void testNoFullValidationErrorReportedWhenFilingDateIsBeforeNextMadeUpToDate() throws ServiceException {
+        companyProfileApi.getConfirmationStatement().setNextMadeUpTo(LocalDate.now().minusDays(11));
+        when(publicDataRetrievalService.getCompanyProfile(ENTITY_NUMBER, PASS_THROUGH_HEADER)).thenReturn(companyProfileApi);
+        updateDto.setFilingDate(LocalDate.now().minusDays(12));
+
+        Errors errors = updateValidator.validateFull(ENTITY_NUMBER, updateDto, new Errors(), LOGGING_CONTEXT, PASS_THROUGH_HEADER);
+        assertFalse(errors.hasErrors());
+    }
+
+    @Test
+    void testFullValidationErrorReportedWhenFilingDateIsAfterNextMadeUpToDate() throws ServiceException {
+        final LocalDate nextMadeUpToDate = LocalDate.now().minusDays(11);
```

Comment:
Could promote minus days value to constant as it is in at least two places.

Reply:
Plus and minus days now all defined as constants and referenced in tests. Now much clearer (I hope!).